cmake/KodiConfig.cmake.in

```diff
@@ -33,4 +33,44 @@ if(NOT CORE_SYSTEM_NAME)
   endif()
 endif()
 
+set(PLATFORM ${CORE_SYSTEM_NAME})
+set(PLATFORM_TAG ${CORE_SYSTEM_NAME})
+
+if(CORE_SYSTEM_NAME STREQUAL windows)
+  include(CheckSymbolExists)
+  check_symbol_exists(_X86_ "Windows.h" _X86_)
+  check_symbol_exists(_AMD64_ "Windows.h" _AMD64_)
+
+  if(_X86_)
+    set(PLATFORM_TAG ${PLATFORM_TAG}-i686)
+  elseif(_AMD64_)
+    set(PLATFORM_TAG ${PLATFORM_TAG}-x86_64)
+  else()
+    message(FATAL_ERROR "Unsupported architecture")
+  endif()
+
+  unset(_X86_)
+  unset(_AMD64_)
+elseif(CORE_SYSTEM_NAME STREQUAL android)
+  if (CPU MATCHES "v7a")
+    set(PLATFORM_TAG ${PLATFORM}-armv7)
+  elseif (CPU MATCHES "arm64")
+    set(PLATFORM_TAG ${PLATFORM}-aarch64)
+  elseif (CPU MATCHES "i686")
+    set(PLATFORM_TAG ${PLATFORM}-i686)
+  else()
+    message(FATAL_ERROR "Unsupported architecture")
+  endif()
+elseif(CORE_SYSTEM_NAME STREQUAL osx)
+  set(PLATFORM_TAG ${PLATFORM}-${CPU})
+elseif(CORE_SYSTEM_NAME STREQUAL ios)
+  if (CPU MATCHES armv7)
+    set(PLATFORM_TAG ${PLATFORM}-armv7)
+  elseif (CPU MATCHES arm64)
+    set(PLATFORM_TAG ${PLATFORM}-aarch64)
+  else()
+    message(FATAL_ERROR "Unsupported architecture")
+  endif()
+endif()
```

Comment:
please move the logic out of this file (i.e. to archsetup). We just want the resulting PLATFORM_TAG here.

Reply:
Moving it to ArchSetup.cmake doesn't work for anything except linux distros (as they use `<PREFIX>/share/kodi/cmake/KodiConfig.cmake` from an already existing Kodi install). All others generate KodiConfig.cmake during building binary add-ons, but that never includes `cmake/scripts/*/ArchSetup.cmake`.

https://github.com/xbmc/xbmc/blob/dd30a2b0723ad97ffb5944d54d6d4a850fca29fe/cmake/scripts/common/PrepareEnv.cmake#L35-L36

Reply:
then move it to PrepareEnv?